parsers/verilog.c

```diff
@@ -1633,9 +1639,25 @@ static int processType (tokenInfo* token, int c, verilogKind* kind)
 		}
 		c = skipDimension (c);
 
+		// break on ',', ';', ')', '}', or other unexpected charactors
 		if (!isWordToken (c))
 			break;
+
+		tokenSaved = dupToken (token);
 		c = readWordToken (token, c);
+		// break on "with"
+		if (token->kind == K_WITH)
+		{
+			// restore tokenSaved to token
+			//   tokenClear() cannot be used to free strings.
+			vStringDelete (token->name);
+			vStringDelete (token->blockName);
+			vStringDelete (token->inheritance);
+			*token = *tokenSaved;
+			eFree (tokenSaved);
```

Comment:
This is too tricky. 
I think you should introduce a helper function doing this dirty work.

```
static void swapToken (tokenInfo *t0, tokenInfo *t1)
{
	tokenInfo tmp = *t0;
	*t0 = *t1;
	*t1 = tmp;
}
...
+ 	if (token->kind == K_WITH)
+ 			swapToken (token, tokenSaved);
```

Reply:
Really Cool!
I looked for any better way, but I could not think of this.
I already merged this request.  I will fix this in the next pull request.

Thanks!